customerio-reactnative.podspec

```diff
@@ -16,6 +16,9 @@ Pod::Spec.new do |s|
   s.source_files = "ios/**/*.{h,m,mm,swift}"
 
   s.dependency "React-Core"
-  s.dependency "CustomerIOTracking", '~> 2.1.0-beta.2'
-  s.dependency "CustomerIOMessagingInApp", '~> 2.1.0-beta.2'
+
+  # Syntax of native iOS pods allows for automatically upgrading to the latest major version of the iOS SDK. 
+  # Reference: https://guides.cocoapods.org/syntax/podfile.html#pod
+  s.dependency "CustomerIO/Tracking", '~> 2'
+  s.dependency "CustomerIO/MessagingInApp", '~> 2'
```

Comment:
Since I have very little knowledge about iOS and its dependencies, small question to understand the change better 🤓

We have seen customers reporting issues with Customer.io dependencies versions mismatch in podfiles and the issues then getting fixed by updating pods. Does the change somehow make their lives better by reducing these errors and not enforcing them to update pods every time? If yes, does this mean there will be lesser customers updating their pod dependencies than before?

I'm assuming pod dependencies are only updated if customers run `pod install --repo-update`. And since they don't get any conflicts, they may not run this every time leaving the old dependencies being used even after updating react native package version?

Please correct me if I'm wrong anywhere in my assumptions 😅

Reply:
This PR fixes errors such as [this one](https://github.com/customerio/customerio-reactnative/issues/125) where the RN SDK wants native version `>= 2.1 && < 2.3` but customers are trying to install `>= 2.3`. 

> I'm assuming pod dependencies are only updated if customers run pod install --repo-update

Actually, no. `pod install` strictly follows the `Podfile.lock` lock file and only installs those pods specified in that file. No updating done. `pod update`, that command will update the dependencies (in the `Podfile.lock`). 

Since you are more familiar with the npm world, `pod install` == `npm ci` (strictly follow lock file). `pod update` == `npm install` (update dependencies including lock file).

Reply:
Thanks for the clarity, this was helpful. The `npm` comparison helped me understand this much better, thank you.